File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/config/TenantContext.java
package com.bittokazi.oauth2.auth.server.config;

public class TenantContext {

    private static ThreadLocal<String> currentTenant = new ThreadLocal<String>();

    private static ThreadLocal<String> currentClient = new ThreadLocal<String>();

    private static ThreadLocal<String> currentDataTenant = new ThreadLocal<String>();

    private static ThreadLocal<String> currentIssuer = new ThreadLocal<String>();

    public static void setCurrentTenant(String tenant) {
        currentTenant.set(tenant);
    }

    public static String getCurrentTenant() {
        return currentTenant.get();
    }

    public static void clear() {
        currentTenant.set(null);
    }

    public static void setCurrentClient(String client) {
        currentClient.set(client);
    }

    public static String getCurrentClient() {
        return currentClient.get();
    }

    public static void clearClient() {
        currentClient.set(null);
    }

    public static void setCurrentIssuer(String client) {
        currentIssuer.set(client);
    }

    public static String getCurrentIssuer() {
        return currentIssuer.get();
    }

    public static void clearCurrentIssuer() {
        currentIssuer.set(null);
    }

    public static void setCurrentDataTenant(String client) {
        currentDataTenant.set(client);
    }

    public static String getCurrentDataTenant() {
        return currentDataTenant.get();
    }

    public static void clearCurrentDataTenant() {
        currentDataTenant.set(null);
    }

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/config/TenantContext.kt
package com.bittokazi.oauth2.auth.server.config

object TenantContext {
    private val currentTenant = ThreadLocal<String?>()

    private val currentClient = ThreadLocal<String?>()

    private val currentDataTenant = ThreadLocal<String?>()

    private val currentIssuer = ThreadLocal<String?>()

    @JvmStatic
    fun setCurrentTenant(tenant: String?) {
        currentTenant.set(tenant)
    }

    @JvmStatic
    fun getCurrentTenant(): String? {
        return currentTenant.get()
    }

    fun clear() {
        currentTenant.set(null)
    }

    fun setCurrentClient(client: String?) {
        currentClient.set(client)
    }

    fun getCurrentClient(): String? {
        return currentClient.get()
    }

    fun clearClient() {
        currentClient.set(null)
    }

    @JvmStatic
    fun setCurrentIssuer(client: String?) {
        currentIssuer.set(client)
    }

    fun getCurrentIssuer(): String? {
        return currentIssuer.get()
    }

    fun clearCurrentIssuer() {
        currentIssuer.set(null)
    }

    @JvmStatic
    fun setCurrentDataTenant(client: String?) {
        currentDataTenant.set(client)
    }

    @JvmStatic
    fun getCurrentDataTenant(): String? {
        return currentDataTenant.get()
    }

    fun clearCurrentDataTenant() {
        currentDataTenant.set(null)
    }
}
